core/src/main/scala/kafka/utils/ShutdownableThread.scala

```diff
@@ -25,6 +25,12 @@ import org.apache.kafka.common.internals.FatalExitError
 abstract class ShutdownableThread(val name: String, val isInterruptible: Boolean = true)
         extends Thread(name) with Logging {
   this.setDaemon(false)
+  setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
+      override def uncaughtException(thread: Thread, exception: Throwable) {
+        error("Exiting due to uncaught exception", exception)
+        Exit.exit(-2)
```

Comment:
does this trigger a shutdown on all other threads in the JVM and let the broker shutdown cleanly?

Reply:
I think so.